quotest/CMakeLists.txt

```
@@ -0,0 +1,9 @@
+# SPDX-FileCopyrightText: 2021 Carl Schwan <carlschwan@kde.org>
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+set(quotest_SRCS quotest.cpp)
+
+add_executable(quotest ${quotest_SRCS})
+add_test(NAME quotest COMMAND quotest)
```

First, starting quotest without parameters will always fail. Something more useful would probably look like:
```suggestion
if (NOT quotest_ROOM)
    quotest_ROOM = "#quotest:matrix.org"
fi
add_test(NAME quotest
               COMMAND quotest ${quotest_USER} ${quotest_PASSWD} ${quotest_DEVICE} ${quotest_ROOM} ${quotest_ORIGIN})
```
Second, I'm not sure if we want to add quotest itself to the list of unit tests - it's comparably long-running (a minute or two), and moreover, it needs actual network access to a Matrix homeserver to do its job. Meanwhile, non-networked build boxes would benefit from being able to just call `ctest` rather than `ctest -E quotest` to run unit tests. What if we just build quotest and then have people run it outside of cmake/ctest framework?

---

I agree adding quotest to ctest doesn't really make sense I removed it.